js/components/welcome.html

```diff
@@ -10,11 +10,23 @@
 				<select data-bind="options:$component.authProviders, optionsText:'name', value:$component.currentAuthProvider"></select> credentials
 				-->
 
-				<p class="text-danger" data-bind="if: isLoggedIn() && tokenExpired()">
+				<p class="text-danger" data-bind="if: isLoggedIn() && tokenExpired() && !$component.isGoogleIapAuth()">
 					Your session has expired. Please login again.
 				</p>
+				<div data-bind="if: isLoggedIn() && tokenExpired() && $component.isGoogleIapAuth()">
+					<p class="text-danger">
+						Your session has expired. Please refresh a page to continue.
+					</p>
 
-				<div>
+					<div class="form-group" data-bind="if: isLoggedIn() && tokenExpired() && $component.isGoogleIapAuth()">
```

Comment:
Why do you repeat `isLoggedIn() && tokenExpired() && $component.isGoogleIapAuth()` which always will be true?